### PR TITLE
adds logic for gcn.notices. notices, replace healpix_file if present

### PR DESCRIPTION
**Pull request by dakota002**

### CODE REVIEW

**Review by lpsinger — CHANGES_REQUESTED** [review]
- [ ] Replace _all_ fields larger than a certain size, not just the `healpix_file` field.
- [ ] Fix CI failures.

**Comment by dakota002** [comment]
> * [ ]  Replace _all_ fields larger than a certain size, not just the `healpix_file` field.
> * [ ]  Fix CI failures.

For the first point, is there a size you have in mind? Like are we talking a max string length?

**Comment by lpsinger** [comment]
> For the first point, is there a size you have in mind? Like are we talking a max string length?

Yes.

**Comment by dakota002** [comment]
> > For the first point, is there a size you have in mind? Like are we talking a max string length?
> 
> Yes.

Any rough idea what the cutoff should be? 256, 1000, ?

**Comment by lpsinger** [comment]
> Any rough idea what the cutoff should be? 256, 1000, ?

Let's start with 512 bytes.

**Review by lpsinger — CHANGES_REQUESTED** [review]
You need to traverse the entire JSON object, not just the top-level dictionary entries.

**Review by lpsinger — CHANGES_REQUESTED** [review]
There are a lot of unrelated, non-functional, stylistic changes here. It looks like you ran a code formatter on this. Please do that in a separate MR that contains only the reformatting.

**Comment by dakota002** [comment]
What are the linting settings for this project? My vscode keeps replacing stuff on save that I don't intend it to

**Comment by lpsinger** [comment]
> What are the linting settings for this project? My vscode keeps replacing stuff on save that I don't intend it to

We don't have any. Tell you what, I'll make a PR to set up [black](https://black.readthedocs.io/en/stable/).

File: gcn_email/core.py [hunk 1]
```diff
@@ -27,10 +28,32 @@
...
+            if isinstance(value, (str, bytes)) and len(value) > max_length:
+                data[key] = REPLACEMENT_TEXT
+            else:
+                replace_long_values(value, max_length)
+    elif isinstance(data, list):
+        for index, item in enumerate(data):
+            if isinstance(item, (str, bytes)) and len(item) > max_length:
+                data[index] = REPLACEMENT_TEXT
+            else:
+                replace_long_values(item, max_length)
```
Comment (lpsinger) [inline gcn_email/core.py:49]: - Refactor duplicate code
- Deserialization of JSON can never produce bytes

```suggestion
    if isinstance(data, dict):
        iter = data.items()
    elif isinstance(data, list):
        iter = enumerate(data)
    else:
        return
    for key, value in iter:
        if isinstance(value, str) and len(value) > max_length:
            data[key] = REPLACEMENT_TEXT
        else:
            replace_long_values(value, max_length)
```

File: gcn_email/core.py [hunk 2]
```diff
@@ -27,10 +28,32 @@
...
+
+
 def get_email_notification_subscription_table():
     client = boto3.client('ssm')
     result = client.get_parameter(
-        Name='/RemixGcnProduction/tables/email_notification_subscription')
+        Name='/RemixGcnProduction/tables/email_notification_subscription'
+    )
```
Comment (lpsinger) [inline gcn_email/core.py:56]: Undo non-functional formatting change.

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please rebase to pick up black formatting.

**Comment by dakota002** [comment]
Some of my changes got removed in the rebase, fixing now

**Review by lpsinger — APPROVED** [review]

**Pull request merged**